cmd/kubeadm/app/phases/upgrade/staticpods.go

```diff
@@ -299,51 +301,95 @@ func performEtcdStaticPodUpgrade(waiter apiclient.Waiter, pathMgr StaticPodPathM
 		return true, fmt.Errorf("fatal error when trying to upgrade the etcd cluster: %v, rolled the state back to pre-upgrade state", err)
 	}
 
-	if isTLSUpgrade {
-		fmt.Printf("[upgrade/etcd] Skipping L7 health-check for %s (as well as data rollback on failure)\n", constants.Etcd)
-	} else {
-		// Checking health state of etcd after the upgrade
-		if _, err = newEtcdCluster.GetStatus(); err != nil {
-			// Despite the fact that upgradeComponent was successful, there is something wrong with etcd cluster
-			// First step is to restore back up of datastore
-			if err := rollbackEtcdData(cfg, fmt.Errorf("etcd cluster is not healthy after upgrade: %v rolling back", err), pathMgr); err != nil {
-				// Even copying back datastore failed, no options for recovery left, bailing out
-				return true, fmt.Errorf("fatal error upgrading local etcd cluster: %v, the backup of etcd database is stored here:(%s)", err, backupEtcdDir)
-			}
-			// Old datastore has been copied, rolling back old manifests
-			if err := rollbackOldManifests(recoverManifests, err, pathMgr, true); err != nil {
-				// Rolling back to old manifests failed, no options for recovery left, bailing out
-				return true, fmt.Errorf("fatal error upgrading local etcd cluster: %v, the backup of etcd database is stored here:(%s)", err, backupEtcdDir)
-			}
-			// Since rollback of the old etcd manifest was successful, checking again the status of etcd cluster
-			if _, err := oldEtcdCluster.GetStatus(); err != nil {
-				// Nothing else left to try to recover etcd cluster
-				return true, fmt.Errorf("fatal error upgrading local etcd cluster: %v, the backup of etcd database is stored here:(%s)", err, backupEtcdDir)
-			}
+	fmt.Println("[upgrade/etcd] waiting for etcd pod to become available after TLS upgrade")
 
-			return true, fmt.Errorf("fatal error upgrading local etcd cluster: %v, rolled the state back to pre-upgrade state", err)
+	// Initialize the new etcd client if it wasn't pre-initialized
+	if newEtcdClient == nil {
+		client, err := etcdutil.NewStaticPodClient(
+			[]string{"localhost:2379"},
+			constants.GetStaticPodDirectory(),
+			cfg.CertificatesDir,
+		)
+		if err != nil {
+			return true, fmt.Errorf("fatal error creating etcd client: %v", err)
+		}
+		newEtcdClient = client
+	}
+
+	// Checking health state of etcd after the upgrade
+	delay := 30 * time.Second
```

Comment:
The delay period is there to prevent the test from passing before the old etcd pod has been stopped by the kubelet. It could potentially be shortened, but that would depend on the kubelet polling interval for static pods and depend on system configuration/speed.

Reply:
Oh, interesting!
This is worth a comment mentioning that not sleeping could race for that reason.

What is the kubelet polling interval?
Should this be a constant or can we retrieve it from somewhere?

Reply:
I added a comment for this, I haven't dug into the kubelet polling interval. I tried to keep this simple for the bandaid backport, but I think using the kubelet api would remove the need for the delay for a more permanent fix.

Reply:
I think having a 30s initial delay regardless of whether `isTLSUpgrade` will probably be fine.

I'd be just as fine initializing it to something low like 2s and increasing it to 30s `if isTLSUpgrade`